src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -54,6 +57,29 @@ def iot_hub_list(client, resource_group_name=None):
         return client.list_by_resource_group(resource_group_name)
 
 
+def iot_hub_update(client, hub_name, parameters, resource_group_name=None):
+    resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
+    return client.create_or_update(resource_group_name, hub_name, parameters, {'IF-MATCH': parameters.etag})
+
+
+def iot_hub_delete(client, hub_name, resource_group_name=None):
+    resource_group_name = _ensure_resource_group_name(client, resource_group_name, hub_name)
+    return client.delete(resource_group_name, hub_name)
+
+
+# Deleting IoT Hub is a long running operation. Due to API implementation issue, 404 error will be thrown during
+# deletion of an IoT Hub.
+# This is a work around to suppress the 404 error. It should be removed after API is fixed.
+class HubDeleteResultTransform(LongRunningOperation):  # pylint: disable=too-few-public-methods
+    def __call__(self, poller):
+        try:
+            super(HubDeleteResultTransform, self).__call__(poller)
+        except CLIError as e:
+            if 'not found' not in str(e):
```

Comment:
Identifying an issue by check if error message contains specific string doesn't seem to be error-prof to me. However, in this case, it seems that the `CLIError` doesn't container enough information for consuming code to determine the source of the issue. I think we should either extend the `CLIError` to enable it to carry more information or introduce a new exception to the hierarchy for HTTP response based errors.

I filed issue https://github.com/Azure/azure-cli/issues/1627 to track this problem. 

This comment of me doesn't intend to block this PR, unless we can find a better solution without new changes to the cli core.